src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,497 @@
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_add_sample_member(context, encode_double_sample_member, &aanderaa_sample.direction_circ_std_rad) != CborNoError) {
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_add_sample_member(context, encode_double_sample_member, &aanderaa_sample.temp_mean_deg_c) != CborNoError) {
+        printf("Failed to add sample member in addSamplesToReport\n");
+      }
+      if (sensor_report_encoder_close_sample(context) != CborNoError) {
+        printf("Failed to close sample in addSamplesToReport\n");
+      }
+      break;
+    }
+    default: {
+      printf("Received invalid sensor type in addSamplesToReport\n");
+      break;
+    }
+  }
+
+}
+
+// Task init
+void reportBuilderInit(cfg::Configuration* sys_cfg) {
+  configASSERT(sys_cfg);
+  _ctx._samplesPerReport = DEFAULT_SAMPLES_PER_REPORT;
+  sys_cfg->getConfig(AppConfig::SAMPLES_PER_REPORT, strlen(AppConfig::SAMPLES_PER_REPORT), _ctx._samplesPerReport);
+  _ctx._transmitAggregations = DEFAULT_TRANSMIT_AGGREGATIONS;
+  sys_cfg->getConfig(AppConfig::TRANSMIT_AGGREGATIONS, strlen(AppConfig::TRANSMIT_AGGREGATIONS), _ctx._transmitAggregations);
+  _ctx._sample_counter = 0;
+  _report_builder_queue =
+      xQueueCreate(REPORT_BUILDER_QUEUE_SIZE, sizeof(report_builder_queue_item_t));
+  configASSERT(_report_builder_queue);
+  // create task
+  BaseType_t rval = xTaskCreate(report_builder_task, "REPORT_BUILDER", 1024, NULL,
+                                REPORT_BUILDER_TASK_PRIORITY, NULL);
+  configASSERT(rval == pdPASS);
+}
+
+// Task
+static void report_builder_task(void *parameters) {
+  (void) parameters;
+
+  report_builder_queue_item_t item;
+
+  while (1) {
+    if (xQueueReceive(_report_builder_queue, &item,
+                      portMAX_DELAY) == pdPASS) {
+      switch (item.message_type) {
+        case REPORT_BUILDER_INCREMENT_SAMPLE_COUNT: {
+          _ctx._sample_counter++;
+          printf("Incrementing sample counter to %d\n", _ctx._sample_counter);
+          if (_ctx._sample_counter >= _ctx._samplesPerReport) {
+            if (_ctx._samplesPerReport > 0 && _ctx._transmitAggregations) {
+              // Need to have more than one node to report anything (1 node would be just the bridge)
+              if (_ctx._report_period_num_nodes > 1) {
+                uint32_t num_sensors = _ctx._report_period_num_nodes - 1; // Minus one since topo includes the bridge
+                sensor_report_encoder_context_t context;
+                uint8_t *cbor_buffer = static_cast<uint8_t *>(pvPortMalloc(MAX_SENSOR_REPORT_CBOR_LEN));
+                configASSERT(cbor_buffer != NULL);
+                sensor_report_encoder_open_report(cbor_buffer, sizeof(cbor_buffer), num_sensors, context);
```

Comment:
We should probably structure this function a bit differently (maybe do/while?) to check the return codes of `sensor_report_encoder_xxx`. This function unlikely fail to right now, but once we start adding more functionality / sensors / etc, we have a chance of failing.

Reply:
Great idea! If we fail to encode part of the message should we still try to send what we were able to encode? (in the case we run out of space in our buffer)

Reply:
Good question, depending on where we fail to encode we could -  but it also may be invalid cbor at that point. I think for now it's probably fine to abort the message on an encoding failure for DVT2, but maybe @towynlin or Evan has more insight on how much of an effort we should make to send things out.

Reply:
Yeah, for DVT2 let's just abort when we're not sure what's happening. With more data gathered in the field, we'll learn what we should handle differently.